src/v2/wallets.ts

```diff
@@ -184,13 +191,28 @@ Wallets.prototype.generateWallet = function(params, callback) {
     const self = this;
     const label = params.label;
 
-    let walletParams = {
+    interface WalletParams {
+      label: any;
+      m: number;
+      n: number;
+      enterprise?: string;
+      disableTransactionNotifications?: boolean;
+      gasPrice?: number;
+      keys?: any[];
+      isCold?: boolean;
+      keySignatures?: any;
+      rootPrivateKey?: any;
+    }
+
+    let walletParams: WalletParams = {
       label: label,
       m: 2,
       n: 3
     };
 
-    if ((!!params.backupXpub + !!params.backupXpubProvider) > 1) {
+    const hasBackupXpub = !!params.backupXpub;
+    const hasBackupXpubProvider = !!params.backupXpubProvider;
+    if ((hasBackupXpub || hasBackupXpubProvider) && hasBackupXpub === hasBackupXpubProvider) {
```

Comment:
Since both of these MUST be boolean after !!-ing them, 
`hasBackupXpub === hasBackupXpubProvider` should be sufficient.

Reply:
This branch catches the condition where BOTH the backup xpub and backup xpub provider parameters are given. These options are mutually exclusive, so if one is true, the other must be false. It's perfectly valid for both to be false, so only checking `hasBackupXpub === hasBackupXpubProvider` would be incorrect.

Reply:
I see. Resolved.

Reply:
Actually I think this is the same as `hasBackupXpub && hasBackupXpubProvider`. Derp. I'll fix.